compiler/rustc_codegen_gcc/src/errors.rs

```diff
@@ -0,0 +1,244 @@
+use rustc_errors::{DiagnosticArgValue, IntoDiagnosticArg};
+use rustc_macros::Diagnostic;
+use rustc_middle::ty::Ty;
+use rustc_span::{Span, Symbol};
+use std::borrow::Cow;
+
+struct ExitCode {
+    pub exit_code: Option<i32>,
+}
+
+impl IntoDiagnosticArg for ExitCode {
+    fn into_diagnostic_arg(self) -> DiagnosticArgValue<'static> {
+        match self.exit_code {
+            Some(t) => t.into_diagnostic_arg(),
+            None => DiagnosticArgValue::Str(Cow::Borrowed("None")),
```

Comment:
Are there circumstances where we actually write "None" out in errors?

Reply:
In this case, no we don't expect the `None` case. Would you suggest adding `unimplemented!()` here?

Reply:
Yeah, I'd prefer a `unimplemented!()` or just changing the type so that it doesn't contain an `Option<..>`.

Reply:
Oh sorry, from looking closer at the docs for [`ExitStatus`](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code) the `None` case happens when the process was terminated by a signal. I think we should leave this case in.

Reply:
Maybe instead of `None` we could write `<signal>` or something in this case that is a little bit more meaningful?